src/generator.rs

```diff
@@ -25,43 +35,109 @@ impl Generator {
             frequency,
             phase: match self.oscillator_state {
                 OscillatorState::Playing { phase, .. } => phase,
+                OscillatorState::Decaying { .. } => 0.0,
```

Comment:
Note: in current monophonic system, resets to phase during decay will trade off audible clicks during decay for having the proper starting phase of the new note.